src/uu/env/src/env.rs

```diff
@@ -165,193 +166,362 @@ pub fn uu_app() -> Command {
-         * environment in weird ways
-         */
-
-        if name.is_empty() {
-            show_warning!("no name specified for value {}", val.quote());
-            continue;
+        // remove all env vars if told to ignore presets
+        if opts.ignore_env {
+            for (ref name, _) in env::vars_os() {
+                env::remove_var(name);
+            }
         }
-        env::set_var(name, val);
-    }
 
-    if opts.program.is_empty() {
-        // no program provided, so just dump all env vars to stdout
-        print_env(opts.line_ending);
-    } else {
-        // we need to execute a command
-        #[cfg(windows)]
-        let (prog, args) = build_command(&mut opts.program);
-        #[cfg(not(windows))]
-        let (prog, args) = build_command(&opts.program);
-
-        /*
-         * On Unix-like systems Command::status either ends up calling either fork or posix_spawnp
-         * (which ends up calling clone). Keep using the current process would be ideal, but the
-         * standard library contains many checks and fail-safes to ensure the process ends up being
-         * created. This is much simpler than dealing with the hassles of calling execvp directly.
-         */
-        match process::Command::new(&*prog).args(args).status() {
-            Ok(exit) if !exit.success() => {
-                #[cfg(unix)]
-                if let Some(exit_code) = exit.code() {
-                    return Err(exit_code.into());
-                } else {
-                    // `exit.code()` returns `None` on Unix when the process is terminated by a signal.
-                    // See std::os::unix::process::ExitStatusExt for more information. This prints out
-                    // the interrupted process and the signal it received.
-                    let signal_code = exit.signal().unwrap();
-                    let signal = Signal::try_from(signal_code).unwrap();
-
-                    // We have to disable any handler that's installed by default.
-                    // This ensures that we exit on this signal.
-                    // For example, `SIGSEGV` and `SIGBUS` have default handlers installed in Rust.
-                    // We ignore the errors because there is not much we can do if that fails anyway.
-                    // SAFETY: The function is unsafe because installing functions is unsafe, but we are
-                    // just defaulting to default behavior and not installing a function. Hence, the call
-                    // is safe.
-                    let _ = unsafe {
-                        sigaction(
-                            signal,
-                            &SigAction::new(SigHandler::SigDfl, SaFlags::empty(), SigSet::all()),
-                        )
-                    };
-
-                    let _ = raise(signal);
+        // load .env-style config file prior to those given on the command-line
+        load_config_file(&mut opts)?;
+
+        // unset specified env vars
```

Comment:
same, explain why we unset these variables.
Security?

Reply:
this is moved code. But I'll address it anyway by extraction to function.
the variables are unset because the corresponding command line argument to unset the variables is provided.